### PR TITLE
switch to xslx and some modernization

### PR DESCRIPTION
**Pull request by btargac**

### CODE REVIEW

File: src/renderer.js [hunk 1]
```diff
@@ -59,7 +59,10 @@ const handleFileSelect = event => {
```
Comment (suleymangunduzz) [inline src/renderer.js:59]: I would await the arrayBuffer here:
`const file = await file.arrayBuffer();`

then use it in the object
`data: {
  ...
  file,
  path: file.path,
}`

Also, do you need a try-catch block for handling the possible errors ?

File: src/utils/generateFileName.spec.js
Comment (suleymangunduzz) [inline src/utils/generateFileName.spec.js:1]: What if you import and mock the original function ?

**Review by suleymangunduzz — APPROVED** [review]

File: src/dialogs.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
 import { dialog } from 'electron';
 
-export const showOpenDialog = (browserWindow, defaultPath, cb) => {
-  dialog.showOpenDialog(browserWindow, {
-    buttonLabel: "Choose",
-    defaultPath,
-    title: "Choose an output folder",
-    properties: ['openDirectory', 'createDirectory']
-  }).then(({ canceled, filePaths }) => {
+export const showOpenDialog = async (browserWindow, data, cb) => {
```
[the remark this answers is not in the document]
Comment (btargac) [reply src/dialogs.js:3]: yeah that makes sense to me actually, I checked why I used that naming; seems electron guys decided to name it that way and I seem to have sticked to it. I've just checked from the electron dialog api documents and they are still happy with it :) their examples are full of `dialog.showOpenDialog`

File: src/renderer.js (hunk 1 above)
Comment (btargac) [reply src/renderer.js:59]: yes much readable than mine, I'll update accordingly 👍🏼  and will check the try catch handler as well

File: src/utils/generateFileName.spec.js
Comment (btargac) [reply src/utils/generateFileName.spec.js:1]: wouldn't we miss the testing the original method's capabilities then ?

**Review by suleymangunduzz — APPROVED** [review]

**Pull request merged**